test/celeritas/field/DormandPrinceStepper.test.hh

```diff
@@ -0,0 +1,227 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020-2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/field/DormandPrinceStepper.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/io/Logger.hh"
+#include "celeritas/field/DormandPrinceMultiStepperGlobal.cuda.hh"
+#include "celeritas/field/DormandPrinceMultiStepperShared.cuda.hh"
+#include "celeritas/field/DormandPrinceStepper.hh"
+#include "celeritas/field/FieldDriver.hh"
+#include "celeritas/field/MagFieldEquation.hh"
+
+namespace celeritas
+{
+namespace test
+{
+//---------------------------------------------------------------------------//
+using celeritas::units::ElementaryCharge;
+using Evaluator_t = celeritas::MagFieldEquation<Real3 (&)(Real3 const&)>;
+using Stepper_uni = celeritas::DormandPrinceStepper<Evaluator_t&>;
+using Stepper_multi_global
+    = celeritas::DormandPrinceMultiStepperGlobal<Evaluator_t&>;
+using Stepper_multi_shared
+    = celeritas::DormandPrinceMultiStepperShared<Evaluator_t&>;
+
+//---------------------------------------------------------------------------//
+// CONSTANTS
+//---------------------------------------------------------------------------//
+constexpr int one_thread = 1;
+constexpr int multi_thread = 4;
+constexpr int number_iterations = 40;
+constexpr OdeState initial_states[5] = {
```

Comment:
It would be interesting to see measurement with more tracks to fill the GPU

Reply:
I use these five states to generate an array of values with a dynamic length ([code here](https://github.com/simbarras/celeritas/blob/7bf84b821c22ba60f9b60ff11793656242a07546/test/celeritas/field/DormandPrinceStepper.test.cu#L158-L164)). The second graph shows the result of the experiment with different number of tracks.